docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md

```diff
@@ -0,0 +1,326 @@
+Started: StepOne         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-a7c77a4e9a02dc4a-01
+Stopped: StepOne         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-a7c77a4e9a02dc4a-01      00:00:00.5093849
+Started: StepTwo         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-9210ad536cae9e4e-01
+Stopped: StepTwo         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-9210ad536cae9e4e-01      00:00:01.0111847
+Stopped: SomeWork        00-bdb5faffc2fc1548b6ba49a31c4a0ae0-c447fb302059784f-01      00:00:01.5236391
+Example work done
+```
+
+Setting <xref:System.Diagnostics.Activity.DefaultIdFormat?displayProperty=nameWithType> is optional
+but helps ensure the sample produces similar output on different .NET runtime versions. .NET 5.0 uses
+the W3C ID format by default but earlier .NET versions default to using 
+<xref:System.Diagnostics.ActivityIdFormat.Hierarchical?displayProperty=nameWithType> as a precaution
+to avoid compatibility issues with older distributed tracing systems. See
+[Activity IDs](distributed-tracing-concepts.md#activity-ids) for more details.
+
+<xref:System.Diagnostics.ActivityListener?displayProperty=nameWithType> is used to receive callbacks
+during the lifetime of an Activity. 
+ - <xref:System.Diagnostics.ActivityListener.ShouldListenTo?displayProperty=nameWithType> - Each
+Activity is associated with an ActivitySource which acts as a namespace for a set of Activities.
+This callback is invoked once for each ActivitySource in the process. Returning true indicates
+the listener should be notified about Activities associated with this source.
+ - <xref:System.Diagnostics.ActivityListener.Sample?displayProperty=nameWithType> - By default
+<xref:System.Diagnostics.ActivitySource.StartActivity?displayProperty=nameWithType> does not
+create an Activity object unless some ActivityListener indicates it should be sampled. Returning
+<xref:System.Diagnostics.ActivitySamplingResult.AllDataAndRecorded?displayProperty=nameWithType>
+indicates that the Activity should be created,
+<xref:System.Diagnostics.Activity.IsAllDataRequested?displayProperty=nameWithType> should be set
+to true, and <xref:System.Diagnostics.Activity.ActivityTraceFlags?displayProperty=nameWithType>
+will have the <xref:System.Diagnostics.ActivityTraceFlags.Recorded?displayProperty=nameWithType>
+flag set. IsAllDataRequested can be observed by the instrumented code as a hint that a listener
+wants to ensure that auxilliary Activity information such as Tags and Events are populated.
+The Recorded flag is encoded in the W3C ID and is a hint to other processes involved in the
+distributed trace that this trace should be logged.
+ - <xref:System.Diagnostics.ActivityListener.ActivityStarted?displayProperty=nameWithType> and
+<xref:System.Diagnostics.ActivityListener.ActivityStopped?displayProperty=nameWithType> are
+called when an Activity is started and stopped respectively. These callbacks provide an
+oportunity to log any relevant information about the Activity. When an Activity has just
```

Comment:
```suggestion
oportunity to add more information (Tags, Events, etc.) to the Activity. When an Activity has just
```

Reply:
Reworded a bit. I want to keep the emphasis on reading from the Activity but I did mention that it is possible to modify it as well.